cupy/core/core.pxd

```diff
@@ -14,6 +14,7 @@ cdef class ndarray:
         readonly object dtype
         readonly memory.MemoryPointer data
         readonly ndarray base
+        readonly object _cuda_array_descr
```

Comment:
Does this feature need cache? Creating `dict` is low cost. And, `dict` is a mutable object. Perhaps, user edit this value though return value of `__cuda_array_interface__`.

Reply:
It saves to 0.3 - 0.5 microseconds per array per kernel launch, which is probably not important unless you are passing many arrays to a function.  I've reverted back to the safer, uncached approach.